src/rust/ensogl/lib/core/src/system/gpu/data/uniform.rs

```diff
@@ -176,20 +176,6 @@ impl<Value:Clone> {
     }
 }}
 
-impl<Value:UniformUpload> UniformData<Value> {
-    /// Uploads the uniform data to the provided location of the currently bound shader program.
-    pub fn upload(&self, context:&Context, location:&WebGlUniformLocation) {
-        self.value.upload_uniform(context,location);
-    }
-}
-
-impl<Value:UniformUpload> Uniform<Value> {
-    /// Uploads the uniform data to the provided location of the currently bound shader program.
-    pub fn upload(&self, context:&Context, location:&WebGlUniformLocation) {
-        self.rc.borrow().upload(context,location)
-    }
-}
```

Comment:
why was it moved to a trait? Now the trait needs to be imported. Why is it better this way?